ante/spamming_prevention.go

```diff
@@ -0,0 +1,118 @@
+package ante
+
+import (
+	"sync"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	oracletypes "github.com/umee-network/umee/x/oracle/types"
+)
+
+// SpammingPreventionDecorator will check if the transaction's gas is smaller than
+// configured hard cap
+type SpammingPreventionDecorator struct {
+	oracleKeeper     OracleKeeper
+	oraclePrevoteMap map[string]int64
+	oracleVoteMap    map[string]int64
+	mu               *sync.Mutex
```

Comment:
Why do we need to pass reference to a mutex here? Can we instead return and use a `*SpamPreventionDecorator`?